src/views/Checkout.vue

```diff
@@ -97,74 +99,109 @@ export default class Checkout extends Vue {
 
         await this.handleOnboardingResult();
 
-        if (this.wallets.length === 0) this.goToOnboarding(true);
-        else this.getBalances();
-    }
-
-    private async getBalances() {
-        const isRefresh = !window.performance || performance.navigation.type === 1;
+        if (this.wallets.length === 0) {
+            this.goToOnboarding(true);
+            return;
+        }
 
-        if (!this.sideResultAddedWallet && this.getLastBalanceUpdateHeight() && !isRefresh) {
-            this.onHeadChange(this.getLastBalanceUpdateHeight()!);
-        } else {
-            // Copy wallets to be able to manipulate them
-            const wallets = this.wallets.slice(0);
-
-            // Generate a new array with references to the respective wallets' accounts
-            const accountsAndContracts = wallets.reduce((acc, wallet) => {
-                acc.push(...wallet.accounts.values());
-                acc.push(...wallet.contracts);
-                return acc;
-            }, [] as Array<AccountInfo | ContractInfo>);
-
-            // Reduce userfriendly addresses from that
-            const addresses = accountsAndContracts.map((accountOrContract) => accountOrContract.userFriendlyAddress);
-
-            // Get balances through pico consensus, also triggers head-change event
-            const network = (this.$refs.network as Network);
-            const balances: Map<string, number> = await network.connectPico(addresses);
-
-            // Update accounts/contracts with their balances
-            // (The accounts are still references to themselves in the wallets' accounts maps)
-            for (const accountOrContract of accountsAndContracts) {
-                const balance = balances.get(accountOrContract.userFriendlyAddress);
-                if (balance === undefined) continue;
-
-                if ('type' in accountOrContract && accountOrContract.type === Nimiq.Account.Type.VESTING) {
-                    // Calculate available amount for vesting contract
-                    accountOrContract.balance = accountOrContract
-                        .calculateAvailableAmount(this.height, Nimiq.Policy.coinsToSatoshis(balance));
+        const balances = await this.getBalances();
+
+        // Handle optional sender address included in the request
+        if (this.request.sender) {
+            let errorMsg = '';
+            // Check if the address exists
+            const senderAddress = this.request.sender.toUserFriendlyAddress();
+            const wallet = this.findWalletByAddress(senderAddress, true);
+            if (wallet) {
+                // Check if that address has enough balance
+                const balance = balances.get(senderAddress);
+                if (balance && Nimiq.Policy.coinsToSatoshis(balance) >= this.minBalance) {
+                    // Forward to Keyguard, skipping account selection
+                    this.setAccountOrContract(wallet.id, senderAddress);
```

Comment:
Does this work on reload?
Wouldn't the `getBalances` be instant in that case and `setAccountOrContract` not execute due to `this.height` being 0?

Reply:
There is not really a possibility for a reload, because if this line gets executed, the user is directly forwarded to the Keyguard. No GUI is displayed in the AM.

Reply:
Furthermore, `getBalances()` calls `this.onHeadChange(cache)` which sets `this.height` anyway.